module/zstd/zfs_zstd.c

```diff
@@ -238,29 +265,16 @@ zstd_mempool_alloc(struct zstd_pool *zstd_mempool, size_t size)
 			 * Check if objects fits the size, if so we take it and
 			 * update the timestamp.
 			 */
-			if (size && !mem && pool->mem && size <= pool->size) {
+			if (pool->mem && size <= pool->size) {
```

Comment:
0 is sometimes passed by zstd and if I read this right now it will allocate something instead of returning NULL.

Reply:
no its never 0 anymore. 0 was a hack for forcing just check reaping of objects. this has been removed entirely now.

Reply:
Ok, I see where it was coming from, my bad. In that case imho it should be asserted the passed value is not 0 though.